### DISCUSSION TITLE
Poll 2 Modbus (energy)meters (SMD72) on one bus

### DISCUSSION THREAD

**Discussion by trollsoft7** [arendst/Tasmota#22360] [Q&A]
Hello, I ran into following problem: I use a ESP32 connected to a Serial/RS485 converter board, which is connected to 2 energy meters (SMD72). I can successfully poll each meter via a script. But I can't achieve to poll both in one script.

`
>D
>B
->sensor53 r
>M 2
+1,21,mN1,0,9600,Licht,25,1,01040500
1,010404ffffffff@i0:1,Import Power,W,power_import,2
+2,21,mN1,0,9600,Heiz,25,1,02040500
2,020404ffffffff@i0:1,Import Power,W,power_import,2
#
`

results in 
<img width="434" alt="image" src="https://github.com/user-attachments/assets/7a2aa09a-3a70-495a-b800-c00624f804e1">
.
Only the second meter is polled. How can I poll both?
Thanks for you help in advance!

**Comment by Pietjuhbel** [comment]
Asuming both devices are on the same bus, did you assign them a seperate node address?

Second, due to the nature of modbus you have to wait for one node to respond in order to poll the second so your script has to be able to work asyncronously.

**Comment by trollsoft7** [comment]
Thanks for your answer! Ah - it might be a needed delay time between polling ID1 and then ID2. Studying the cookbooks today I am unsure where to insert the delay. Do you have a link to an example?
Thanks!

**Comment by Pietjuhbel** [comment]
Wel, the problem with modbus is that the 2 wire version is strictly half duplex so a delay would not necessarily help you a lot, if your devices are very consistent in time it takes them to respond you could probably get away with it though.
I am not very familiar with scripting in tasmota / berry, I use a node server that does the handling of all the data polling and exchange.
My sollution to this problem is to have the query to the next device be triggered by the response of the first and visa versa. This is way easier to implement in javascript than in a script on the ESP

**Reply by gemu2015** [reply]
there are several examples on multiple devices on one bus in the sml docs like PZEM004T
